src/freerct.cpp

```diff
@@ -125,6 +125,16 @@ int freerct_main(int argc, char **argv)
 		                "medium-path = /usr/share/fonts/gnu-free/FreeSans.ttf\n");
 		return 1;
 	}
+	/* Overwrite the default language settings if the user specified a custom language in the config file. */
+	const char *preferred_language = cfg_file.GetValue("language", "language");
+	if (preferred_language != nullptr) {
+		int index = GetLanguageIndex(preferred_language);
+		if (index < 0) {
+			fprintf(stderr, "Warning: Language '%s' is not known, ignoring\n");
```

Comment:
Missing a parameter for the `%s`.

In addition, in its current form, the message is useless information. It currently basically just says "it doesn't work", which is really old news for a user that tries to set or change a language.

Never just say "something fails", always try to provide sufficient information such that a user can find the cause and fix it. Here you would say you read the language name from the configuration file.

Further extensions on this path are stating the file path of that file, listing the currently known languages, try a case-insensitive check, try a fuzzy match to give a hint to the use ("maybe you meant UVW language?"), and more.

It may be more work, but you'll make many users happy, get fewer help requests into your inbox, and eventually even you won't remember where the message is coming from, and you'll be glad for the provided assistance.